### ISSUE TITLE
Error 429 on big helm chart repo

### ISSUE THREAD

**Issue by startxfr** [artifacthub/hub#3900]
**Describe the bug**
It's been 2 days, when artifacthub process my helm-repo, it get some 429 error on file that always exist. 

**To Reproduce**
Have a look at the last sync of the repo abe79b96-e489-4c1b-b47e-4e93f41fcb3a for complete list

```
error getting logo image https://helm-repository.readthedocs.io/en/latest/img/cluster-rbac.svg: unexpected status code received: 429 (package: cluster-rbac version: 14.6.323)
error getting logo image https://helm-repository.readthedocs.io/en/latest/img/cluster-redis.svg: unexpected status code received: 429 (package: cluster-redis version: 14.6.323)
error getting logo image https://helm-repository.readthedocs.io/en/latest/img/cluster-router.svg: unexpected status code received: 429 (package: cluster-router version: 14.6.323)
error getting logo image https://helm-repository.readthedocs.io/en/latest/img/cluster-sso.svg: unexpected status code received: 429 (package: cluster-sso version: 14.6.323)
error getting logo image https://helm-repository.readthedocs.io/en/latest/img/cluster-storage-efs.svg: unexpected status code received: 429 (package: cluster-storage-efs version: 14.6.323)
error getting logo image https://helm-repository.readthedocs.io/en/latest/img/cluster-storage.svg: unexpected status code received: 429 (package: cluster-storage version: 14.6.323)
error getting logo image https://helm-repository.readthedocs.io/en/latest/img/cluster-vault.svg: unexpected status code received: 429 (package: cluster-vault version: 14.6.323)
error getting logo image https://helm-repository.readthedocs.io/en/latest/img/cluster-vault.svg: unexpected status code received: 429 (package: cluster-vault-config version: 14.6.323)
...
error getting logo image https://helm-repository.readthedocs.io/en/latest/img/operator.svg: unexpected status code received: 429 (package: operator version: 14.6.323)
error getting logo image https://helm-repository.readthedocs.io/en/latest/img/project.svg: unexpected status code received: 429 (package: project version: 14.6.323)
error getting logo image https://helm-repository.readthedocs.io/en/latest/img/sxapi.svg: unexpected status code received: 429 (package: sxapi version: 14.6.323)
error getting repository metadata: unexpected status code received: 429
```

**Expected behavior**
only 200 response like before

**Desktop (please complete the following information):**
 - Server side between artifacthub and readthedocs

**Additional context**
Readthedocs doesn't inform us about any change in their filtering policy. I try to use it during artifacthub sync period, no 429 error for me. 

Is it possible to introduce a parameter in the configuration of the repository to control the rythm of the sync (adding some sleep or interval between calls)

I don't really know how I can solve my sync problem on the abe79b96-e489-4c1b-b47e-4e93f41fcb3a repository. Do you have any sugestion ?

Many thanks for your support (and artifacthub services ;)

**Comment by tegioz** [comment]
Hi @startxfr 👋

That `429` status code most likely means that we are being rate limited. I've just processed your repo and I started seeing those errors in just a few seconds.

The level of concurrency while processing a repository cannot be controlled per repository. At the moment we process concurrently 10 chart versions per repository, and a number of repositories are processed concurrently as well. Out of curiosity, I've checked what would happen processing only one chart version at a time in your repository, but I still got rate limited pretty quickly. It's still too fast.

Repositories are expected to be fully processed in a few minutes, otherwise we interrupt the process (as we need to process thousands of them quite often). Introducing configurable pauses would slow everything down, potentially causing some other problems, so we'd rather not to do that.

I've noticed that you release a new version for all your charts at once, which is fine, but it forces us to do a lot of requests as well 😇

The only solution I can think of at the moment would be to host your charts somewhere else (or maybe just your logos, if all the errors reported are related to them). It looks like there are no other repositories registered in `artifacthub.io` using the `readthedocs.io` domain (some may be using the same service but with a custom domain though).

Hope this helps a bit 🙂
